web-registry/src/components/molecules/ProjectSelect.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { useState, useEffect } from 'react';
+import { Field, useFormikContext } from 'formik';
+import { startCase } from 'lodash';
+
+import SelectTextField, {
+  Option,
+} from 'web-components/lib/components/inputs/SelectTextField';
+import { useCreditClassByOnChainIdQuery } from '../../generated/graphql';
+
+interface FieldProps {
+  creditClassId: string;
+  name?: string;
+  required?: boolean;
+}
+
+const defaultProjectOption = { value: '', label: 'Choose Project' };
+
+const ProjectSelect: React.FC<FieldProps> = ({
+  creditClassId,
+  name = "metadata['regen:vcsProjectId']",
+  required,
+  ...props
+}) => {
+  const { data: dbDataByOnChainId } = useCreditClassByOnChainIdQuery({
+    variables: { onChainId: creditClassId as string },
+  });
+  const { setFieldValue } = useFormikContext();
+  const [projectOptions, setProjectOptions] = useState<Option[]>([]);
+
+  useEffect(() => {
+    setProjectOptions([]);
+    if (creditClassId === '') {
+      // reset options when creditClassID no longer has a selected element
+      return;
+    }
+    // reset project if creditClassID changes
+    setFieldValue(name, '');
+    const projects =
+      dbDataByOnChainId?.creditClassByOnChainId?.projectsByCreditClassId?.nodes;
+
+    const dbOptions =
+      projects?.map(project => {
+        const projectId = project?.metadata?.['regen:vcsProjectId']?.['@value'];
```

Comment:
FYI we're updating this to be just this instead:
```suggestion
        const projectId = project?.metadata?.['regen:vcsProjectId'];
```

ref: https://github.com/regen-network/regen-web/pull/959 and https://github.com/regen-network/regen-registry-ops/pull/7#issuecomment-1130562234

Reply:
thx